packages/angular/projects/clr-angular/src/forms/common/providers/control-class.service.ts

```diff
@@ -7,17 +7,25 @@
 import { Injectable, Optional, Renderer2 } from '@angular/core';
 import { LayoutService } from './layout.service';
 
+const CLASS_ERROR = 'clr-error';
+const CLASS_SUCCESS = 'clr-success';
+
 @Injectable()
 export class ControlClassService {
   className = '';
 
   constructor(@Optional() private layoutService: LayoutService) {}
 
-  controlClass(invalid = false, grid = false, additional = '') {
+  // @TODO: rework how we pass arguments
+  controlClass(invalid = false, valid = false, grid = false, additional = '') {
```

Comment:
It may be ... not sure yet :)

I think as a good start will be to changing from multiple arguments into a single object as a way to pass data inside the function. It won't require any more to know the exact order of the arguments, it also will let us be more descriptive with what exactly is true or false.

This is clearly visible into the tests
```ts
controlClass(false, true, false)
```

I think that this is more readable 
```ts
controlClass({ invalid: false, valid: true, grid: false })
```
A bit more code but it's clear what it does without the need of guessing.

Not sure yet how the change of control state will have effects on this method.